cwms-data-api/src/main/java/cwms/cda/data/dto/timeseriesprofile/TimeSeriesProfile.java

```diff
@@ -0,0 +1,165 @@
+package cwms.cda.data.dto.timeseriesprofile;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import cwms.cda.api.errors.FieldException;
+import cwms.cda.data.dto.CwmsDTO;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV2;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class)
+@JsonDeserialize(builder = TimeSeriesProfile.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public final class TimeSeriesProfile extends CwmsDTO
+{
+	@Schema(description = "Location ID")
+	private final String locationId;
+	@Schema(description = "Description")
+	private final String description;
+	@Schema(description = "Parameter List")
+	private final List<String> parameterList;
+	@Schema(description = "Key Parameter")
+	private final String keyParameter;
+	@Schema(description = "Reference TS")
+	private final String refTsId;
+
+	private TimeSeriesProfile(Builder builder)
+	{
+		super(builder.officeId);
+		this.locationId = builder.locationId;
+		this.description = builder.description;
+		this.keyParameter = builder.keyParameter;
+		this.parameterList = builder.parameterList;
+		this.refTsId = builder.refTsId;
+	}
+
+	public String getLocationId()
+	{
+		return locationId;
+	}
+
+	public String getDescription()
+	{
+		return description;
+	}
+
+	public String getKeyParameter()
+	{
+		return keyParameter;
+	}
+
+	public List<String> getParameterList()
+	{
+		return parameterList!=null ? new ArrayList<>(parameterList) : null;
+	}
+
+	public String getRefTsId()
+	{
+		return refTsId;
+	}
+
+	@Override
+	public void validate() throws FieldException
+	{
+		if (this.parameterList == null) {
+			throw new FieldException("Parameter list field can't be null");
```

Comment:
FYI, there are more specific exceptions derived from this exception that can be used.

Reply:
changed to use the Validator